packages/react-hooks/src/ElementUtils.ts

```diff
@@ -7,10 +12,13 @@ import { InferComponentProps } from '@deephaven/utils';
  * @param type The type to check against
  * @returns True if the node is a React element of the specified type
  */
-export function isElementOfType<T>(
-  node: ReactNode,
-  type: T
-): node is ReactElement<InferComponentProps<T>> {
+export function isElementOfType<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  T extends string | JSXElementConstructor<any> =
```

Comment:
Could have `string | JSXElementConstructor<any>` defined as a type above to make this a little nicer.